sunpy/physics/differential_rotation.py

```diff
@@ -87,317 +90,94 @@ def diff_rot(duration, latitude, rot_type='howard', frame_time='sidereal'):
-
-    # ... and the semi-diameter
-    # Form the mean anomalies of Venus(MV),Earth(ME),Mars(MM),Jupiter(MJ)
-    # and the mean elongation of the Moon from the Sun(D).
-    t = de / 36525.0
-    mv = 212.60 + np.mod(58517.80 * t, 360.0)
-    me = 358.4760 + np.mod(35999.04980 * t, 360.0)
-    mm = 319.50 + np.mod(19139.860 * t, 360.0)
-    mj = 225.30 + np.mod(3034.690 * t, 360.0)
-    d = 350.70 + np.mod(445267.110 * t, 360.0)
-
-    # Form the geocentric distance(r) and semi-diameter(sd)
-    r = 1.0001410 - (0.0167480 - 0.00004180 * t) * np.cos(np.deg2rad(me)) \
-        - 0.000140 * np.cos(np.deg2rad(2.0 * me)) \
-        + 0.0000160 * np.cos(np.deg2rad(58.30 + 2.0 * mv - 2.0 * me)) \
-        + 0.0000050 * np.cos(np.deg2rad(209.10 + mv - me)) \
-        + 0.0000050 * np.cos(np.deg2rad(253.80 - 2.0 * mm + 2.0 * me)) \
-        + 0.0000160 * np.cos(np.deg2rad(89.50 - mj + me)) \
-        + 0.0000090 * np.cos(np.deg2rad(357.10 - 2.0 * mj + 2.0 * me)) \
-        + 0.0000310 * np.cos(np.deg2rad(d))
-
-    sd_const = constants.radius / constants.au
-    sd = np.arcsin(sd_const / r) * 10800.0 / np.pi
-
-    return {"p": Angle(p, u.deg),
-            "b0": Angle(b, u.deg),
-            "sd": Angle(sd.value, u.arcmin),
-            "l0": Angle(0.0, u.deg)}
-
-
-def _sun_pos(date):
-    """
-    Calculate solar ephemeris parameters.  Allows for planetary and lunar
-    perturbations in the calculation of solar longitude at date and various
-    other solar positional parameters. This routine is a truncated version of
-    Newcomb's Sun and is designed to give apparent angular coordinates (T.E.D)
-    to a precision of one second of time.  This function replicates the SSW/
-    IDL function "sun_pos.pro".  This function is assigned to be
-    internal at the moment as it should really be replaced by accurate
-    ephemeris calculations in the part of SunPy that handles ephemeris.
-
-    Parameters
-    -----------
-    date : `sunpy.time.time`
-        Time at which the solar ephemeris parameters are calculated.  The
-        input time can be in any acceptable time format.
-
-    Returns
-    -------
-    A dictionary with the following keys with the following meanings:
-
-    longitude  -  Longitude of sun for mean equinox of date (degs)
-    ra         -  Apparent RA for true equinox of date (degs)
-    dec        -  Apparent declination for true equinox of date (degs)
-    app_long   -  Apparent longitude (degs)
-    obliq      -  True obliquity (degs)
+    # Compute Stonyhurst Heliographic co-ordinates - returns (longitude,
+    # latitude). Points off the limb are returned as nan.
+    heliographic_coordinate = coordinate.transform_to('heliographic_stonyhurst')
 
-    Notes
-    -----
-    SSWIDL code equivalent:
-        http://hesperia.gsfc.nasa.gov/ssw/gen/idl/solar/sun_pos.pro
+    # Compute the differential rotation
+    drot = diff_rot(interval, heliographic_coordinate.lat.to(u.degree), **diff_rot_kwargs)
+
+    # Set the observer
```

Comment:
I think with the latest changes to #2186 you don't need this block at all. You can just pass the observer string or baseframe or skycoord into the `SkyCoord` constructor along with the new obstime and it will do the rest.